rp2040-hal/Cargo.toml

```diff
@@ -18,57 +18,35 @@ targets = ["thumbv6m-none-eabi"]
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
+bitfield = { version = "0.14.0" }
+chrono = { version = "0.4", default-features = false, optional = true }
 cortex-m = "0.7.2"
-embedded_hal_0_2 = { package = "embedded-hal", version = "0.2.5", features = [
-  "unproven",
-] }
+critical-section = { version = "1.0.0" }
+defmt = { version = ">=0.2.0, <0.4", optional = true }
+embedded_hal_0_2 = { package = "embedded-hal", version = "0.2.5", features = ["unproven"] }
+embedded-dma = "0.2.0"
 embedded-hal = "1.0.0"
-embedded-hal-nb = "1.0.0"
 embedded-hal-async = "1.0.0"
-embedded-dma = "0.2.0"
+embedded-hal-nb = "1.0.0"
 embedded-io = "0.6.1"
+frunk = { version = "0.4.1", default-features = false }
 fugit = "0.3.6"
+i2c-write-iter = { version = "1.0.0", features = ["async"], optional = true }
 itertools = { version = "0.10.1", default-features = false }
 nb = "1.0"
-rp2040-pac = { version = "0.6.0", features = ["critical-section"] }
 paste = "1.0"
 pio = "0.2.0"
+rand_core = "0.6.3"
 rp2040-hal-macros = { version = "0.1.0", path = "../rp2040-hal-macros" }
+rp2040-pac = { version = "0.6.0", features = ["critical-section"] }
+rtic-monotonic = { version = "1.0.0", optional = true }
 usb-device = "0.3"
 vcell = "0.1"
 void = { version = "1.0.2", default-features = false }
```

Comment:
As we’re changing the deps order in the manifest, I think it’d be nice to have the optional dependencies separated from the mandatory ones to give them more visibility.

Reply:
Done